rust/agents/validator/src/submit.rs

```diff
@@ -58,17 +58,22 @@ impl ValidatorSubmitter {
         }
     }
 
+    /// Submits signed checkpoints.
+    /// If a target checkpoint is provided, it must not be bethind the provided tree, and
```

Comment:
I wonder if we should just have two separate functions for when the target is provided vs. not. That imo would make the code more readable